### PR TITLE
Fix full cookie restriction

### PR DESCRIPTION
**Pull request by astroash**
Related to #634

#### Changes
* Hacky fix for redirect on full admin access routes

#### User testing flow:
- log into visitor app
- manually go to a `/cb/...` route
- every page should redirect to `ConfirmPassword` page

### CODE REVIEW

File: src/cb_admin/pages/Activities.js [hunk 1]
```diff
@@ -87,7 +87,8 @@ export default class ActivitiesPage extends React.Component {
   }
 
   componentDidMount() {
-    Promise.all([Activities.get(), CommunityBusiness.getActivities()])
+    CommunityBusiness.update() // used to check cookie permissions
+      .then(Promise.all([Activities.get(), CommunityBusiness.getActivities()]))
```
Comment (eliascodes) [inline src/cb_admin/pages/Activities.js]: Needs to be a function

File: src/cb_admin/pages/Dashboard.js [hunk 2]
```diff
@@ -50,9 +50,10 @@ export default class Dashboard extends React.Component {
   }
 
   componentDidMount() {
-    CommunityBusiness.get({ fields: ['name'] })
+    CommunityBusiness.update() // used to check cookie permissions
+      .then(CommunityBusiness.get({ fields: ['name'] }))
```
Comment (eliascodes) [inline src/cb_admin/pages/Dashboard.js]: Needs to be a function

File: src/cb_admin/pages/Visitor.js [hunk 3]
```diff
@@ -114,11 +114,12 @@ export default class VisitorProfile extends React.Component {
   }
 
   componentDidMount() {
-    Promise.all([
-      Visitors.get({ id: this.props.match.params.id }),
-      Visitors.genders(),
-      CommunityBusiness.get({ fields: ['name', 'logoUrl'] }),
-    ])
+    CommunityBusiness.update() // used to check cookie permissions
+      .then(Promise.all([
```
Comment (eliascodes) [inline src/cb_admin/pages/Visitor.js]: Needs to be a function

File: src/cb_admin/pages/VisitorDetails.js [hunk 4]
```diff
@@ -104,19 +104,12 @@ export default class VisitorDetailsPage extends React.Component {
   }
 
   componentDidMount() {
-    Visitors.genders()
+    CommunityBusiness.update() // used to check cookie permissions
+      .then(Visitors.genders())
```
Comment (eliascodes) [inline src/cb_admin/pages/VisitorDetails.js]: Needs to be a function

**Comment by astroash** [comment]
> There's a problem with the transformResponse function. I get an error saying cannot read property result of undefined.

You get this error if your mock does not return a payload with `{ result: ...}`.

File: src/cb_admin/pages/Dashboard.js (hunk 2 above)
Comment (astroash) [reply src/cb_admin/pages/Dashboard.js]: What do you mean?

Comment (eliascodes) [reply src/cb_admin/pages/Dashboard.js]: The value passed to `then` must be a function. You're passing a `Promise`.

Comment (astroash) [reply src/cb_admin/pages/Dashboard.js]: Oh yeah...

Weird that this works for the `Activities` page

**Comment by astroash** [comment]
@eliascodes soz ended up force pushing over your commit as there was a lot of overlap

🙏🏽 squash and merge

File: src/cb_admin/pages/VisitorDetails.js [hunk 5]
```diff
@@ -104,19 +104,12 @@ export default class VisitorDetailsPage extends React.Component {
   }
 
   componentDidMount() {
-    Visitors.genders()
+    CommunityBusiness.update() // used to check cookie permissions
+      .then(Visitors.genders)
```
Comment (eliascodes) [inline src/cb_admin/pages/VisitorDetails.js]: Needs to be a function

File: src/cb_admin/pages/__tests__/Dashboard.test.js [hunk 6]
```diff
@@ -35,6 +38,14 @@ describe('Dashboard Component', () => {
   test(':: unsuccessful response with 401 redirects to login', async () => {
     expect.assertions(1);
 
+    mock.onPut('/community-businesses/me', {
+      name: undefined,
```
Comment (eliascodes) [inline src/cb_admin/pages/__tests__/Dashboard.test.js]: Does this defo need the `undefined`s?

Comment (astroash) [reply src/cb_admin/pages/__tests__/Dashboard.test.js]: Nope, I can remove

File: src/cb_admin/pages/VisitorDetails.js (hunk 5 above)
Comment (eliascodes) [reply src/cb_admin/pages/VisitorDetails.js]: Oh, oops, I've just noticed that this is a function

**Comment by eliascodes** [comment]
Soz, ignore those comments, but I'll still wait till I can user test this tonight before merging.

File: src/cb_admin/pages/VisitorDetails.js (hunk 5 above)
Comment (astroash) [reply src/cb_admin/pages/VisitorDetails.js]: but it *is* a function...

**Comment by eliascodes** [comment]
User tested 👍

**Comment by astroash** [comment]
😭 what happened to squash and merge?

**Comment by eliascodes** [comment]
Ah crap, sorry I was on autopilot

**Pull request merged**